include/kernel/faulthandler.h

```diff
@@ -7,11 +7,29 @@
 #pragma once
 
 #include <object.h>
+#ifdef CONFIG_KERNEL_MCS
+#include <kernel/sporadic.h>
+#endif
 
 #ifdef CONFIG_KERNEL_MCS
 static inline bool_t validTimeoutHandler(tcb_t *tptr)
 {
-    return cap_get_capType(TCB_PTR_CTE_PTR(tptr, tcbTimeoutHandler)->cap) == cap_endpoint_cap;
+    cap_t handlerCap = TCB_PTR_CTE_PTR(tptr, tcbTimeoutHandler)->cap;
```

Comment:
I think this is ok. 

Food for thought: maybe we could still set the thread to a timeout faulted state, but not deliver the fault anywhere. this would allow timeout fault handlers to 'recover' from missed faults.

Reply:
We'd need to add some way of getting that state from the thread so we could check it later but that's entirely possible. I don't think a well configured system should be able to enter this state though.